Software/tft/header/firmware.go

```diff
@@ -27,6 +33,24 @@ func firmwareImageSize(c *Config) (uint32, error) {
 
 }
 
+// ReadHeader tries to read header from input file
+func ReadHeader(c *Config) (*Header, error) {
+	buf := make([]byte, 16)
+	n, err := c.InputFile.Read(buf)
+	if err != nil {
+		return nil, err
+	}
+	if n == 16 {
```

Comment:
Maybe make it a constant and use in like 38?

Reply:
What happens if `n != 16`? seems like we should return an error?

Reply:
I agree, will patch in a second